clients/notification/src/test/java/org/eclipse/hono/notification/deviceregistry/DevicesOfTenantDeletedNotificationTest.java

```diff
@@ -0,0 +1,106 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.notification.deviceregistry;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import java.time.Instant;
+
+import org.eclipse.hono.notification.AbstractNotification;
+import org.eclipse.hono.notification.NotificationConstants;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.vertx.core.json.Json;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Tests verifying the behavior of {@link DevicesOfTenantDeletedNotification}.
+ *
+ */
+public class DevicesOfTenantDeletedNotificationTest {
+
+    private static final String CREATION_TIME = "2007-12-03T10:15:30Z";
+    private static final String TENANT_ID = "my-tenant";
+
+    private AbstractNotification notification;
+
+    /**
+     * Sets up the notification.
+     */
+    @BeforeEach
+    public void setUp() {
+        notification = new DevicesOfTenantDeletedNotification(TENANT_ID, Instant.parse(CREATION_TIME));
+    }
+
+    /**
+     * Verifies that the expected properties are contained in the JSON.
+     */
+    @Test
+    public void testThatValuesAreContainedInJson() {
+
+        final JsonObject json = JsonObject.mapFrom(notification);
+
+        assertThat(json.getString(NotificationConstants.JSON_FIELD_SOURCE))
+                .isEqualTo(NotificationConstants.SOURCE_DEVICE_REGISTRY);
+        assertThat(json.getInstant(NotificationConstants.JSON_FIELD_CREATION_TIME).toString()).isEqualTo(CREATION_TIME);
+
+        assertThat(json.getString(NotificationConstants.JSON_FIELD_TYPE))
+                .isEqualTo(DevicesOfTenantDeletedNotification.TYPE);
+        assertThat(json.getString(NotificationConstants.JSON_FIELD_TENANT_ID)).isEqualTo(TENANT_ID);
+    }
+
+    /**
+     * Verifies that the serialization did not change: the JSON contains the static values and only the expected keys.
+     */
+    @Test
+    public void testThatSerializationIsStable() {
+
+        final JsonObject json = JsonObject.mapFrom(notification);
+        assertThat(json).isNotNull();
+
+        // When adding new properties to the data object, make sure not to break the existing API because messages might
+        // be persisted. For breaking changes add a new object mapper class instead.
+        final int expectedPropertiesCount = 4;
+        assertWithMessage("JSON contains unknown fields").that(json.size()).isEqualTo(expectedPropertiesCount);
+
+        assertThat(json.getString("type")).isEqualTo("devices-of-tenant-deleted-v1");
+        assertThat(json.getString("source")).isEqualTo("device-registry");
+        assertThat(json.getString("creation-time")).isEqualTo(CREATION_TIME);
+
+        assertThat(json.getString("tenant-id")).isNotNull();
```

Comment:
How about using static constants instead of string literals as in the other tests? Example: NotificationConstants.JSON_FIELD_TENANT_ID instead of _tenant-id_?

Reply:
That would contradict the intention of this test. There is already another test that checks that the JSON contains the values of the constants. This test is intended to detect when the constant is (accidentally) changed. This could lead to the fact that serialized (i.e. in Kafka stored) notifications cannot be de-serialized again.